analyses/mutation-signatures/util/mut_sig_functions.R

```diff
@@ -0,0 +1,204 @@
+  #
+  #   which_sig_list: a list of `whichSignature` output
+  #   muts_per_sample: a vector with the total mutation signature counts for each 
+  #                    sample as calculated by deconstructSigs::mut.to.sigs.input 
+  #   wgs_genome_size: size of the WGS genome in bp
+  #   wxs_genome_size: size of the WXS exome in bp
+  #   maf_df: a data.frame with `short_histology` and `experimental strategy`  
+  #           information columns
+  #
+  # Returns:
+  #   A data.frame that is samples x signatures and has the number of mutations
+  #   per mb for each signature x sample combo
+
+  # Count the total number of signature mutations for each sample
+  total_muts <- apply(sigs_input, 1, sum)
+
+  # Pull out the signature weights and make into matrix
+  sig_num_df <- do.call(
+    "rbind.data.frame",
+    lapply(which_sig_list, function(sample_data) sample_data$weights)
+  ) %>%
+    tibble::rownames_to_column("Tumor_Sample_Barcode") %>%
+
+  # Calculate the number of mutations contributing to each signature
+  # Here the weight is multiplied by the total number of signature mutations. 
+  dplyr::mutate_at(dplyr::vars(-Tumor_Sample_Barcode), ~ . * total_muts) %>%
+    as.data.frame() %>%
+    
+    # Join the short_histology and experimental stategy information
+    dplyr::inner_join(dplyr::select(
+      maf_df,
+      "Tumor_Sample_Barcode",
+      "short_histology",
+      "experimental_strategy"
+    ),
+    by = "Tumor_Sample_Barcode"
+    ) %>%
+
+    # Get rid of Panel samples
+    dplyr::filter(experimental_strategy != "Panel") %>%
+    
+    # Reformat for plotting
+    reshape2::melt() %>%
+
+    # Only keep distinct
+    dplyr::distinct(Tumor_Sample_Barcode, variable, .keep_all = TRUE) %>%
+
+    # Add genome size and calculate the mutation per this column
+    dplyr::mutate(
+      genome_size = dplyr::recode(experimental_strategy,
+        "WGS" = wgs_size,
+        "WXS" = wxs_size
+      ),
+      mut_per_mb = value / (genome_size / 10^6)
+    ) %>% 
+    dplyr::rename(num_mutations = value, 
+                  signature = variable)
+  
+  return(sig_num_df)
+}
+
+bubble_matrix_plot <- function(sig_num_df, label = "none") {
+  # Given the data.frame output from `calc_mut_per_sig` that has the number of 
+  # mutations per Mb that belong to each sample x signature combo, make a bubble 
+  # matrix plot that displays this information for all samples but summarized by 
+  # histology.
+  #
+  # Args:
+  #
+  #   sig_num_df: a data.frame with number of mutations per Mb that belong to
+  #               each sample x signature combo
+  #   label: a character string for the title of the plot to be passed to ggplot2::ggtitle
+  #
+  # Returns:
+  #   A bubble matrix plot with the number of mutations per Mb and proportion of 
+  #   tumors with a non-zero weight for all samples summarized by histology.
+  #
+  # Summarize the mut_per_mb column by histology
+  grouped_sig_num <- sig_num_df %>%
+    dplyr::arrange(signature) %>%
```

Comment:
This is what you will presumably need to fix to make these numerically ordered, rather than alphabetical. Though you also shouldn't need to order the df until after grouping and summarizing.

Reply:
Now that you mention it, I am unsure what this `arrange` is here for.